include/dbstructure.php

```diff
@@ -154,6 +158,19 @@ function update_structure($verbose, $action, $tables=null, $definition=null) {
 	if (is_null($definition))
 		$definition = db_definition($charset);
 
+	// Ensure index conversion to unique removes duplicates
+	$sql_config = "SET session old_alter_table=1;";
+	if ($verbose)
+		echo $sql_config."\n";
+	if ($action)
+		@$db->q($sql_config);
+
+	// MySQL >= 5.7.4 doesn't support the IGNORE keyword in ALTER TABLE statements
+	if (version_compare($db->getdb()->server_info, '5.7.4') >= 0) {
```

Comment:
Theoretically we support the old mysql functionality as well (and not only mysqli). This would fail when mysqli wasn't used. Could you add a check for it? You could use the class variable $mysqli (boolean) for it.